src/CodeGen_LLVM.cpp

```diff
@@ -1460,6 +1473,34 @@ void CodeGen_LLVM::codegen(Stmt s) {
     s.accept(this);
 }
 
+Type CodeGen_LLVM::upgrade_type_for_arithmetic(Type t) const {
```

Comment:
nit: wouldn't it be better to pass Type by const-ref?

Reply:
done